designs/wahid/src/front.js

```diff
@@ -273,22 +259,40 @@ export default (part) => {
         90
       )
     }
-
     // Facing/Lining boundary (flb)
     paths.flbFacing = new Path()
       .move(points.dartTop)
       .curve(points.flbCp, points.flbCpTop, points.flbTop)
       .attr('class', 'fabric')
     paths.flbLining = paths.flbFacing.clone().attr('class', 'lining dashed')
 
+    // Grainline
+    if (options.hemStyle === 'classic') {
+      points.grainLineFromFront = new Point(points.hemTip.x, points.closureTop.y)
+      points.grainLineToFront = new Point(points.hemTip.x, points.hemTip.y).shift(90, 20)
+    }
+    if (options.hemStyle === 'rounded') {
+      points.grainLineFromFront = new Point(points.roundEnd.x, points.roundEnd.y).shift(90, 200)
+      points.grainLineToFront = new Point(points.roundEnd.x, points.roundEnd.y)
+    }
+    if (options.hemStyle === 'square') {
+      points.grainLineFromFront = new Point(points.cutonfoldTo.x, points.cutonfoldTo.y).shift(
```

Comment:
When using square hemStyle, the grainline can extend beyond the pattern. You might want to think about having the endpoint based on points.closuretop.y.

![Screen Shot 2022-08-13 at 7 05 45 PM](https://user-images.githubusercontent.com/109869956/184519584-60597f43-7c5c-40e2-946e-f155b45263dd.png)